### ISSUE TITLE
TypeError: Object.hasOwn is not a function

### ISSUE THREAD

**Issue by trend-derek-hawker** [cheton/styled-ui#775]
Hi, I think you can close this issue right away. Just wanted to document it in case anyone else comes across it.

We had an issue recently where AWS canaries began failing after a recent upgrade of Tonic. We could not reproduce the error manually or in other frameworks like Cypress. We did some more digging and saw the following error in the logs

```
Error: TypeError: Object.hasOwn is not a function
    at https://dev-cloudone.trendmicro.com/_landing/static-1687465167169/js/vendors~c1c-web-app-landing.1a4680e5.chunk.js:9466:19
    at Array.filter (<anonymous>)
    ...
```

The latest version of syn-nodejs-puppeteer-4.0 uses Chromium 92, which doesn't have support for `hasOwn` that was used in [a recent PR](https://github.com/trendmicro-frontend/tonic-ui/pull/771/files#diff-ffa161e1dc23bd0b655d1adbfe4360f0952724b3936512201d50609ba03a219eR120). Related [Can I Use](https://caniuse.com/mdn-javascript_builtins_object_hasown).

Like I said, not something that needs fixing as it can be solved by adding a babel polyfill. I just wanted to share in case others comes across this issue. I was surprised how old the version of Chromium is in AWS Synthetic Canaries.

**Comment by cheton** [comment]
It seems that the usage of `Object.prototype.hasOwnProperty.call(obj, prop)` is still present in several places within the codebase. See the following occurrences:

```sh
$ grep -r "hasOwnProperty" packages/*/src
packages/react/src/color-style/useColorStyle.js:  if (!Object.prototype.hasOwnProperty.call(colorStyle, colorMode)) {
packages/styled-system/src/sx.js:    if (!Object.prototype.hasOwnProperty.call(styleProps, key)) {
packages/styled-system/src/utils/merge-object.js:    if (!Object.prototype.hasOwnProperty.call(a, key)) {
packages/styled-system/src/core/parser.js:    if (!Object.prototype.hasOwnProperty.call(raw, key)) {
```

To address compatibility issues in certain environments and ensure consistency in the codebase, I have created a pull request (#776) to revert the change from `Object.hasOwn(obj, prop)` back to `Object.prototype.hasOwnProperty.call(obj, prop)`. This change will maintain compatibility and align with the previous usage.

**Comment by cheton** [comment]
Fixed in [@tonic-ui/styled-system@1.8.1](https://github.com/trendmicro-frontend/tonic-ui/releases/tag/%40tonic-ui%2Fstyled-system%401.8.1) and [@tonic-ui/react@1.21.2](https://github.com/trendmicro-frontend/tonic-ui/releases/tag/%40tonic-ui%2Freact%401.21.2)

**Comment by trend-derek-hawker** [comment]
Sounds good. Not sure it would be a bad thing to move toward use of `hasOwn` over `hasOwnProperty` given the improvements over edge cases. The canaries are running on a 2.5 year old version of Chromium which is pretty ancient at this point. 

I found it wasn't too hard to get this working after adding polyfills on webpack+babel  projects. 

Just needed to have `import 'core-js/stable'` in your entry file and the following .babelrc worked. The .babelrc was already set up this way, but pretty sure it's the `"@babel/preset-env"` that is the important part.

```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": "defaults",
        "useBuiltIns": "entry",
        "corejs": "3.20.0",
        "modules": "auto"
      }
    ],
  ],
  "plugins": [
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-nullish-coalescing-operator"
  ]
}
```
